pkg/controller/networkpolicy/networkpolicy_controller.go

```diff
@@ -476,7 +497,9 @@ func (n *NetworkPolicyController) createAddressGroup(namespace string, podSelect
 	}
 	klog.V(2).Infof("Creating new AddressGroup %s with selector (%s)", addressGroup.Name, addressGroup.Selector.NormalizedName)
 	n.addressGroupStore.Create(addressGroup)
-	n.groupingInterface.AddGroup(addressGroupType, addressGroup.Name, groupSelector)
+	if nodeSelector == nil {
```

Comment:
Please add comment for this condition: "For an AddressGroup that selects Nodes via nodeSelector, we calculate its members via NodeLister directly, instead of groupingInterface which handles Pod and ExternalEntity currently."

Reply:
Done.